python/src/otel/otel_sdk/otel-instrument

```diff
@@ -1,44 +1,160 @@
+    environ[PYTHONPATH] += os.pathsep + os.environ[LAMBDA_RUNTIME_DIR]
+
+# Configure OpenTelemetry Python with environment variables
 
-# config default traces exporter if missing
-environ.setdefault("OTEL_TRACES_EXPORTER", "otlp_proto_grpc_span")
+# - Set Lambda Layer python packages in current python path so we can find them
+#   right away in this script
 
-# set service name
-if environ.get("OTEL_RESOURCE_ATTRIBUTES") is None:
-    environ["OTEL_RESOURCE_ATTRIBUTES"] = "service.name=%s" % (
-        environ.get("AWS_LAMBDA_FUNCTION_NAME")
+if LAMBDA_LAYER_PKGS_DIR not in sys.path:
+    sys.path.append(LAMBDA_LAYER_PKGS_DIR)
+
+from opentelemetry.environment_variables import (
+    OTEL_PROPAGATORS,
+    OTEL_TRACES_EXPORTER,
+    OTEL_PYTHON_DISABLED_INSTRUMENTATIONS,
+)
+from opentelemetry.sdk.environment_variables import OTEL_RESOURCE_ATTRIBUTES
+
+# - Set the default Trace Exporter
+
+environ.setdefault(OTEL_TRACES_EXPORTER, "otlp_proto_grpc_span")
+
+# - Set the service name
+
+service_name_resource_attribute = "service.name"
+if environ.get(OTEL_RESOURCE_ATTRIBUTES) is None:
+    environ[OTEL_RESOURCE_ATTRIBUTES] = "%s=%s" % (
+        service_name_resource_attribute,
+        environ.get(AWS_LAMBDA_FUNCTION_NAME),
     )
-elif "service.name=" not in environ.get("OTEL_RESOURCE_ATTRIBUTES"):
-    environ["OTEL_RESOURCE_ATTRIBUTES"] = "service.name=%s,%s" % (
-        environ.get("AWS_LAMBDA_FUNCTION_NAME"),
-        environ.get("OTEL_RESOURCE_ATTRIBUTES"),
+elif service_name_resource_attribute not in environ.get(
+    OTEL_RESOURCE_ATTRIBUTES
+):
+    environ[OTEL_RESOURCE_ATTRIBUTES] = "%s=%s,%s" % (
+        service_name_resource_attribute,
+        environ.get(AWS_LAMBDA_FUNCTION_NAME),
+        environ.get(OTEL_RESOURCE_ATTRIBUTES),
     )
 
-# TODO: Remove if sdk support resource detector env variable configuration.
+# - Set the Resource Detectors (Resource Attributes)
+#
+#   TODO: waiting on OTel Python support for configuring Resource Detectors from
+#   an environment variable. Replace the bottom code with the following when
+#   this is possible.
+#
+#   environ["OTEL_RESOURCE_DETECTORS"] = "aws_lambda"
+#
 lambda_resource_attributes = (
     "cloud.region=%s,cloud.provider=aws,faas.name=%s,faas.version=%s"
     % (
         environ.get("AWS_REGION"),
-        environ.get("AWS_LAMBDA_FUNCTION_NAME"),
+        environ.get(AWS_LAMBDA_FUNCTION_NAME),
         environ.get("AWS_LAMBDA_FUNCTION_VERSION"),
     )
 )
-environ["OTEL_RESOURCE_ATTRIBUTES"] = "%s,%s" % (
+environ[OTEL_RESOURCE_ATTRIBUTES] = "%s,%s" % (
     lambda_resource_attributes,
-    environ.get("OTEL_RESOURCE_ATTRIBUTES"),
+    environ.get(OTEL_RESOURCE_ATTRIBUTES),
 )
 
-# start the runtime with the extra options
-system(" ".join(args))
+# - Set the default propagators
+
+environ.setdefault(OTEL_PROPAGATORS, "tracecontext,b3,xray")
+
+# - Disable Lambda instrumentation because we will instrument when we know it
```

Comment:
We should just remove the auto instrumentation entry point from setup.cfg instead as it can't really be used